debugScripts/debugFailTx.js

```diff
@@ -0,0 +1,421 @@
+            return false;
+        }
+    }
+
+    txHash = process.argv[3];
+
+    if (!web3.utils.isHex(txHash)){
+        console.log("Illegal tx Hash input (not hex): " + txHash);
+        return false;
+    }
+
+    if (txHash.length != 66){
+        console.log("Illegal tx Hash input: " + txHash + ",  wrong size: " + txHash.length);
+        return false;
+    }
+
+}
+
+function printHelp () {
+    console.log("usage: \'node debugFailTx network txHash\'.");
+    console.log("network options: m (mainnet) / r (ropsten) / k (kovan).");
+}
+
+haveTokenInst = 0;
+
+async function checkBalances(tokenAdd, traderAdd, blockNumber, tradeQty, msgEthValue, isBuy) {
+    let errString = '';
+
+    let abi = solcOutput.contracts["MockERC20.sol:MockERC20"].interface;
+    tokenInst = await new web3.eth.Contract(JSON.parse(abi), tokenAdd);
+
+    // check to see if token balance changed in this block. if it did. its trade success.
+    txData = await tokenInst.methods.balanceOf(traderAdd).encodeABI();
+    let tokenBalance = web3.utils.toBN(await web3.eth.call({to:tokenAdd, data:txData}, blockNumber));
+    let prevTokenBalance = web3.utils.toBN(await web3.eth.call({to:tokenAdd, data:txData}, (blockNumber - 1)));
+
+    if (tokenBalance.cmp(prevTokenBalance) == 0) {
+        errString += "Trade failed. ";
+    }
+
+    if (isBuy) {
+        let userBalance = web3.utils.toBN('0x' + await web3.eth.getBalance(traderAdd, blockNumber).valueOf());
+        let prevBalance = web3.utils.toBN('0x' + await web3.eth.getBalance(traderAdd, (blockNumber - 1)).valueOf());
+//        console.log("current balance: " + userBalance.toString(16) + "\nprev block balance: " + prevBalance.toString(16));
+        if(tradeQty.gt(prevBalance)) errString += " user eth balance: " + prevBalance.toString(16) + " < Trade qty: " + tradeQty.toString(16);
+        if(msgEthValue.cmp(tradeQty) != 0) errString += (" Trade qty not equal msg value.");
+        //make sure reserve has enough tokens.
+
+    } else {
+        let txData = await tokenInst.methods.allowance(traderAdd, kyberNetworkAdd).encodeABI();
+        let allowance = web3.utils.toBN(await web3.eth.call({to:tokenAdd, data:txData}, (blockNumber)));
+
+        if (msgEthValue != 0) errString += "Bad msg value: " + msgEthValue + " expected 0";
+        if(tradeQty.gt(allowance)) errString += " Allowance: " + allowance + " < trade qty: " + tradeQty;
+
+        //use token balance of previous block. this blocks balance reflects operations in this block as well.
+        if (tradeQty.gt(prevTokenBalance)) errString += " Token balance: " + prevTokenBalance.toString(16) +  " < Trade qty: " + tradeQty.toString(16);
+    }
+
+    return errString;
+}
+
+async function checkMinConversionRate(blockNumber, src, dest, tradeQty, minConversionRate) {
+    let errString = '';
+
+    let abi = solcOutput.contracts["KyberNetwork.sol:KyberNetwork"].interface;
+    NetWorkInst = await new web3.eth.Contract(JSON.parse(abi), kyberNetworkAdd);
```

Comment:
'NetWorkInst' - so many wrong naming conventions in a single variable name...

Reply:
Ha
I am glad someone can appreciate my accomplishments.